examples/vanilla-es6/src/model.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Creates a new Model instance and hooks up the storage.
+ * @constructor
+ * @param {object} storage A reference to the client side storage class
+ */
+export default class Model {
+	constructor(storage) {
+		this.storage = storage;
+	}
+
+	/**
+	 * Creates a new todo model
+	 *
+	 * @param {string} [title] The title of the task
+	 * @param {function} [callback] The callback to fire after the model is created
+	 */
+	create(title, callback){
+		title = title || '';
+
+		let newItem = {
+			title: title.trim(),
+			completed: false
+		};
+
+		this.storage.save(newItem, callback);
+	}
+
+	/**
+	 * Finds and returns a model in storage. If no query is given it'll simply
+	 * return everything. If you pass in a string or number it'll look that up as
+	 * the ID of the model to find. Lastly, you can pass it an object to match
+	 * against.
+	 *
+	 * @param {string|number|object} [query] A query to match models against
+	 * @param {function} [callback] The callback to fire after the model is found
+	 *
+	 * @example
+	 * model.read(1, func) // Will find the model with an ID of 1
+	 * model.read('1') // Same as above
+	 * //Below will find a model with foo equalling bar and hello equalling world.
+	 * model.read({ foo: 'bar', hello: 'world' })
+	 */
+	read(query, callback){
+		const queryType = typeof query;
+
+		if (queryType === 'function') {
+			callback = query;
```

Comment:
Please don't reassign parameters. In this case you could just pass it to `findAll` anyway, right?